src/BlockchainWriter/Router.ts

```diff
@@ -50,12 +51,14 @@ export class Router {
     }
   }
 
-  blockDownloadedConsumer = async (blockDownloaded: BlockDownloaded): Promise<void> => {
+  blockDownloadedConsumer = async (blockDownloaded: BlockDownloaded): Promise<void> =>
     await this.claimController.setBlockInformationForTransactions(
       getTxnIds(blockDownloaded.poetBlockAnchors),
       blockDownloaded.block,
     )
-  }
+
+  onPurgeStaleTransactions = async (): Promise<void> =>
+    await this.claimController.purgeStaleTransactions()
```

Comment:
So we want to do this in the router instead of the controller? Is there a reason for that?

Reply:
Yes.

Keeps code simpler and achieves the same results, IMO.

We're doing it at the router because that's as far up as we can go right now, but in the future we could do it at the root of the module, after we refactor the RMQ Client code. Kind of catching an application-wide error.

Reply:
ok, so catch as high up the food chain as possible. gotcha

Reply:
Also: unless we have a guarantee that `purgeStaleTransactions()` will never throw, I can't recall exactly right now but I think I saw some weird / unwanted things going on if errors got thrown inside a callback passed to ampqlib. Need to test that again to refresh my mind.

Reply:
well there was a catch that didn't re-throw down in the controller, so that should not have been an issue. But I do understand the idea to catch as high as possible.